DQMOffline/Trigger/plugins/ZGammaplusJetsMonitor.cc

```diff
@@ -0,0 +1,813 @@
+#include "DQMOffline/Trigger/plugins/TriggerDQMBase.h"
+#include "CommonTools/Utils/interface/StringCutObjectSelector.h"
+#include "DataFormats/JetReco/interface/PFJet.h"
+#include "DataFormats/JetReco/interface/PFJetCollection.h"
+#include "DataFormats/Common/interface/TriggerResults.h"
+#include "DataFormats/HLTReco/interface/TriggerEvent.h"
+#include "DataFormats/HLTReco/interface/TriggerObject.h"
+#include "DataFormats/HLTReco/interface/TriggerTypeDefs.h"
+#include "HLTrigger/HLTcore/interface/HLTConfigProvider.h"
+#include "FWCore/Utilities/interface/EDGetToken.h"
+#include "FWCore/Common/interface/TriggerNames.h"
+#include "DataFormats/Math/interface/deltaR.h"
+
+#include "TLorentzVector.h"
+#include <cassert>
+#include "TPRegexp.h"
+
+class ZGammaplusJetsMonitor : public DQMEDAnalyzer, public TriggerDQMBase {
+public:
+  typedef dqm::reco::MonitorElement MonitorElement;
+  typedef dqm::reco::DQMStore DQMStore;
+
+  ZGammaplusJetsMonitor(const edm::ParameterSet&);
+  ~ZGammaplusJetsMonitor() throw() override;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+protected:
+  void bookHistograms(DQMStore::IBooker&, edm::Run const&, edm::EventSetup const&) override;
+  void analyze(edm::Event const& iEvent, edm::EventSetup const& iSetup) override;
+  void dqmBeginRun(edm::Run const& iRun, edm::EventSetup const& iSetup) override;
+
+  bool islowRefPt(double RefPt);
+  bool ismediumRefPt(double RefPt);
+  bool ishighRefPt(double RefPt);
+
+  bool isBarrel(double eta);
+  bool isEndCapInner(double eta);
+  bool isEndCapOuter(double eta);
+  bool isForward(double eta);
+
+  bool isMatched(double hltJetEta, double hltJetPhi, double OffJetEta, double OffJetPhi);
+
+  void bookMESub(DQMStore::IBooker&,
+                 ObjME* a_me,
+                 const int len_,
+                 const std::string& h_Name,
+                 const std::string& h_Title,
+                 const std::string& h_subOptName,
+                 const std::string& h_subOptTitle,
+                 const bool doDirectBalancevsReferencePt = true,
+                 const bool bookDen = false);
+
+  void FillME(ObjME* a_me,
+              const double directbalance_,
+              const double Difjetref_,
+              const double Assymetry_,
+              const double ReferencePt_,
+              const double Jetpt_,
+              const bool doDirectBalancevsReferencePt = true);
+
+private:
+  const std::string folderName_;
+
+  const std::string processName_;  // process name of (HLT) process for which to get HLT configuration
+  // The instance of the HLTConfigProvider as a data member
+  HLTConfigProvider hltConfig_;
+
+  edm::EDGetTokenT<trigger::TriggerEvent> triggerEventObject_;
+  edm::EDGetTokenT<edm::TriggerResults> triggerResultsToken_;
+  const std::string pathName;
+  const std::string moduleName;
+  edm::InputTag jetInputTag_;
+  edm::EDGetTokenT<reco::PFJetCollection> jetToken_;
```

Comment:
```suggestion
  const edm::EDGetTokenT<reco::PFJetCollection> jetToken_;
```

as it is in the initializer list.